File: BeeCrowd/ListaPapainoel/Noel.java
package ListaPapainoel;
import java.util.*;

public class Noel {
    
    public static void ordena(List<String> listas){

        for(int i = 1; i < listas.size(); i++){
            String temp = listas.get(i); 
            int j = i - 1;

            while(j >= 0 && listas.get(j).compareTo(temp) > 0){

                listas.set(j + 1, listas.get(j));
                j--;
            }
            listas.set(j + 1, temp);

        }


    }

    public static void main(String[] args) {
        
        Scanner ler = new Scanner(System.in);

        List<String> lista = new ArrayList<>();

        int tam = ler.nextInt();
        int comportou = 0;
        int baguncou = 0;

        for(int i = 0; i < tam; i++){
            
            String comp = ler.next();
            String name = ler.next();

            if(comp.equals("+")){
                comportou++;
            }else if(comp.equals("-")){
                baguncou++;
            }
            lista.add(name);

        }

        ordena(lista);

        for(int i = 0; i < lista.size();i++){
            System.out.println(lista.get(i));
        }
        System.out.println("Se comportaram: " + comportou + " | Nao se comportaram: " + baguncou);


        ler.close();
    }

}

File: BeeCrowd/ListaPapainoel/Noel.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

void ordena(char *array[], int tam)
{
    char temp[20];

    for (int i = 1; i < tam; i++)
    {
        strcpy(temp, array[i]);
        int j = i - 1;

        while (j >= 0 && strcmp(array[j], temp) > 0)
        {

            strcpy(array[j + 1], array[j]);
            j--;
        }
        strcpy(array[j + 1], temp);
    }
}

int main()
{

    int tam;
    int comportou = 0;
    int baguncou = 0;

    scanf("%i", &tam);

    char **array = (char **)malloc(tam * sizeof(char *));

    for (int i = 0; i < tam; i++)
    {

        array[i] = (char *)malloc(20 * sizeof(char));

        char comp;
        scanf(" %c", &comp);
        scanf(" %s", array[i]);

        if (comp == '+')
        {
            comportou++;
        }
        else if (comp == '-')
        {
            baguncou++;
        }
    }

    ordena(array, tam);

    for (int i = 0; i < tam; i++)
    {
        printf("%s\n", array[i]);
    }
    printf("Se comportaram: %i | Nao se comportaram: %i\n", comportou, baguncou);

    for (int i = 0; i < tam; i++)
    {
        free(array[i]);
    }
    free(array);

    return 0;
}
